python/sqlflow_submitter/xgboost/dataset.py

```diff
@@ -131,8 +195,13 @@ def load_dmatrix(filename):
     in detailed.
     '''
     if xgb.rabit.get_world_size() > 1:
+        if '#' in filename:
```

Comment:
Please add description here what `if '#' in filename` means.

Reply:
'#' in XGBoost means use external memory cache. But we use `load_svmlight_files` to load DMatrix in distributed mode. '#' is not accepted by `load_svmlight_files`. Please see comment [here](https://github.com/sql-machine-learning/sqlflow/pull/2361/files#diff-85436bed613ae85f0caae0da7b565094R198).